gplugins/design_recipe/DesignRecipe.py

```diff
@@ -1,81 +1,151 @@
 from __future__ import annotations
 
-from typing import List, Optional
+import json
+import os
+import pickle
+from collections.abc import Callable
+from pathlib import Path, PosixPath, WindowsPath
+
+import pydantic
 from gdsfactory import Component
-from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.config import logger
 from gdsfactory.path import hashlib
+from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.typings import ComponentFactory
+from pydantic import BaseModel
 
 import gplugins.design_recipe as dr
 
 
+class Setup:
+    """
+    A dynamic class to store any setup information
```

Comment:
setup of what? a Simulation? "Setup" feels like too generic a name even if this is a base class

Reply:
Mainly used to store DesignRecipe setup info, but it can be used to store arbitrary simulation setup info